action.yml

```diff
@@ -191,6 +200,45 @@ runs:
       set +x
     shell: bash
 
+  - name: Log the next step action
+    run: echo ▷ Figuring out change detection and coverage
+    shell: bash
+  - name: Determine change detection and coverage
+    id: compute-change-detection-coverage
+    run: |
+      # Compute whether to use change detection and coverage
+      import json
+      import os
+      import pathlib
+      import sys
+
+      FILE_APPEND_MODE = 'a'
+      OUTPUTS_FILE_PATH = pathlib.Path(os.environ['GITHUB_OUTPUT'])
+
+      def set_output(name, value):
+          with OUTPUTS_FILE_PATH.open(FILE_APPEND_MODE) as outputs_file:
+              outputs_file.writelines(f'{name}={value}{os.linesep}')
+
+      # Input from GHA
+      pull_request_change_detection = json.loads('${{
+        inputs.pull-request-change-detection
+      }}')
+      pull_request_branch = '${{ github.event.pull_request.base.ref || '' }}'
```

Comment:
I've been looking at the source code, and it seems that the branch naming may bring surprises. It seems like `ansible-test` only allows this to be `devel` or `stable-*`. This will probably explode on non community-maintained collections...

Reply:
@mattclay could you comment on this? Should we allow non-core repos to be more flexible on the branch naming?

Reply:
I guess. But since this action also supports Ansible 2.9, it'd mean the lack of feature parity, so it'd still need some sort of shim.
Also, could you clarify what exactly do you expect to be moved to `ansible-test`? Fake branch creation? Reading some env var info?

Reply:
@mattclay are you saying that the change detection is also intended to be used with push events and not just pull requests?

Reply:
Yes, change detection is used for two different scenarios in CI:

* Pull Requests - Based on files changed in the PR.
* Commits - Based on files changed since the most recent commit which passed CI. This results in changes accumulating across multiple commits when those commits are either untested or did not pass CI.

Reply:
While `--changed` is also passed for pushes on AZP, it seems to always run *all* tests. I checked out a few runs for https://github.com/ansible-collections/community.general/commit/6a03108609cba48b88b57d2bf43eb238fc13f229, CI already passed for the previous commit in `main`, and for this commit also all tests were run.

Reply:
@felixfontein I spotted this warning in the output for that CI job:

`WARNING: No successful commit found. All tests will be executed.`

That's why all tests were executed. I'll have to take a look to see if I can figure out why that happened.

Reply:
Interesting. I think the current behavior (run all tests for pushes) has been this way for quite some time now.

Reply:
It's working properly for ansible/ansible. I'll have to look to see if the issue is with all collections, or just community.general.

Reply:
I saw the same errors for community.crypto, community.docker, and community.network:
- https://dev.azure.com/ansible/community.crypto/_build/results?buildId=59540&view=logs&j=bdbef642-b100-5732-d0c4-b572d7493d07&t=13973663-e317-5e59-f498-514d4b141b35&l=200
- https://dev.azure.com/ansible/community.docker/_build/results?buildId=59539&view=logs&j=df593170-0476-56b6-4a33-d8e2815d43d8&t=ed5b4094-b5e7-57c7-f236-602faf856c9f&l=183
- https://dev.azure.com/ansible/community.network/_build/results?buildId=60698&view=logs&j=16b3c408-a84a-5a59-0453-baf9c22a8e18&t=89323c31-06d0-5f15-540e-edfa0e5d23e8&l=288

Reply:
Thanks for the input! So it seems like it's fine to have a temporary workaround in the action until such a time that `ansible-test` supports it natively.